From aeefd49fe0cc374921420f9022528ff3782f408e Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Fri, 20 Apr 2018 17:08:47 +0200
Subject: [PATCH 0597/1051] phy: cp110-comphy: add support for USB host comphy
 initialization

The USB comphy configuration is performed by firmware, therefore the
comphy driver only triggers appropriate smc call with argument understood
by fw.

Change-Id: I71bfecc8e33c8ee040f843d8c484de20800317cf
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/phy/marvell/phy-mvebu-cp110-comphy.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
index 3f32d7500ef4..ed3ee78f7d09 100644
--- a/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
+++ b/drivers/phy/marvell/phy-mvebu-cp110-comphy.c
@@ -85,11 +85,13 @@ static const struct mvebu_comhy_conf mvebu_comphy_cp110_modes[] = {
 	MVEBU_COMPHY_CONF(1, 2, PHY_MODE_2500SGMII),
 	MVEBU_COMPHY_CONF(1, 0, PHY_MODE_PCIE),
 	MVEBU_COMPHY_CONF(1, 0, PHY_MODE_SATA),
+	MVEBU_COMPHY_CONF(1, 0, PHY_MODE_USB_HOST),
 	/* lane 2 */
 	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_SGMII),
 	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_2500SGMII),
 	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_10GKR),
 	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_PCIE),
+	MVEBU_COMPHY_CONF(2, 0, PHY_MODE_USB_HOST),
 	/* lane 3 */
 	MVEBU_COMPHY_CONF(3, 1, PHY_MODE_SGMII),
 	MVEBU_COMPHY_CONF(3, 1, PHY_MODE_2500SGMII),
@@ -101,6 +103,7 @@ static const struct mvebu_comhy_conf mvebu_comphy_cp110_modes[] = {
 	MVEBU_COMPHY_CONF(4, 0, PHY_MODE_10GKR),
 	MVEBU_COMPHY_CONF(4, 1, PHY_MODE_SGMII),
 	MVEBU_COMPHY_CONF(4, 1, PHY_MODE_PCIE),
+	MVEBU_COMPHY_CONF(4, 1, PHY_MODE_USB_HOST),
 	/* lane 5 */
 	MVEBU_COMPHY_CONF(5, 2, PHY_MODE_SGMII),
 	MVEBU_COMPHY_CONF(5, 2, PHY_MODE_2500SGMII),
@@ -180,6 +183,11 @@ static int mvebu_comphy_power_on(struct phy *phy)
 				 lane->id,
 				 COMPHY_FW_MODE_FORMAT(COMPHY_SATA_MODE));
 		break;
+	case PHY_MODE_USB_HOST:
+		ret = comphy_smc(MV_SIP_COMPHY_POWER_ON, priv->phys,
+				 lane->id,
+				 COMPHY_FW_MODE_FORMAT(COMPHY_USB3H_MODE));
+		break;
 	default:
 		return -ENOTSUPP;
 	}
-- 
2.17.1

